source/shared/core_conn.cpp

```diff
@@ -282,7 +362,8 @@ SQLRETURN core_odbc_connect( _Inout_ sqlsrv_conn* conn, _Inout_ std::string& con
         SQLCHAR state[ SQL_SQLSTATE_BUFSIZE ];
         SQLSMALLINT len;
         SQLRETURN sr = SQLGetDiagField( SQL_HANDLE_DBC, conn->handle(), 1, SQL_DIAG_SQLSTATE, state, SQL_SQLSTATE_BUFSIZE, &len );
-        // sql state IM002/IM003 in ODBC 17, means that the correct ODBC driver is not installed
+        // sql state IM002, means that the specified ODBC driver is not installed
+        // might return IM003 in ODBC 17 (an issue that will be fixed)
         is_missing_driver = ( SQL_SUCCEEDED(sr) && state[0] == 'I' && state[1] == 'M' && state[2] == '0' && state[3] == '0' && (state[4] == '2' || state[4] == '3'));
```

Comment:
why do we need to compare character by character instead of the whole string?

Reply:
good question...